api/src/main/java/org/apache/iceberg/types/CheckCompatibility.java

```diff
@@ -32,13 +32,27 @@
 public class CheckCompatibility extends TypeUtil.CustomOrderSchemaVisitor<List<String>> {
   /**
    * Returns a list of compatibility errors for writing with the given write schema.
+   * This includes the required/optional compatibility of fields and returns error in
+   * case of mismatch.
```

Comment:
It isn't clear what "required/optional compatibility" is. What about this instead?

> This includes nullability: writing a optional (nullable) values to a required field is an error.

Reply:
Your wording is better, I modified accordingly. 😄